nilearn/image/image.py

```diff
@@ -728,6 +728,12 @@ def new_img_like(ref_niimg, data, affine=None, copy_header=False):
         if isinstance(ref_niimg, nibabel.freesurfer.mghformat.MGHImage):
             default_dtype = np.uint8
         data = as_ndarray(data, dtype=default_dtype)
+    if data.dtype in (np.int64, np.uint64):
```

Comment:
Do we want to raise a warning here to tell user that the `dtype` has been changed?

Reply:
thanks, I was going to ask for opinions about what should be the behavior here.
I believe that because int64 is the representation chosen by numpy when it is given python ints, this downcasting will happen very frequently. And as there probably very few use cases (if any) for images containing ints that don't fit in 32 bits, for the huge majority of users, it will not make a difference. So I am hesitant to emit a warning. Moreover, I noted that in the case the data contains `bool` (the few lines just above), the casting is done silently (but in that case it is a safe conversion).

At the same time, silently doing an unsafe type conversion is of course not a good idea in principle.

Reply:
I feel we should have a warning.